app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningAudioFeedback.kt

```diff
@@ -0,0 +1,56 @@
+package com.github.HumanLearning2021.HumanLearningApp.view
+
+import android.content.Context
+import android.media.MediaPlayer
+import com.github.HumanLearning2021.HumanLearningApp.R
+
+class LearningAudioFeedback(private val applicationContext : Context) {
+    private lateinit var correctMp: MediaPlayer
+    private lateinit var incorrectMp: MediaPlayer
+
+    fun initMediaPlayers(){
+        correctMp = MediaPlayer.create(applicationContext, R.raw.bravo)
+        incorrectMp = MediaPlayer.create(applicationContext, R.raw.dommage)
+    }
+
+    fun releaseMediaPlayers(){
+        // we release when the user pauses the so that we don't eat up resources uselessly
+        correctMp.release()
+        incorrectMp.release()
+    }
+
+
+    fun stopAndPrepareMediaPlayers(){
+        stopAndPrepare(correctMp)
+        stopAndPrepare(incorrectMp)
+    }
+
+    private fun stopAndPrepare(mp : MediaPlayer) {
+        if (mp.isPlaying) {
+            mp.stop()
+            // from https://developer.android.com/reference/android/media/MediaPlayer#prepare()
+            // "For files, it is OK to call prepare(), which blocks until MediaPlayer
+            // is ready for playback."
+            mp.prepare()
+        }
+    }
+
+    fun startCorrectFeedback(){
+        correctMp.start()
+    }
+
+    fun startIncorrectFeedback(){
+        incorrectMp.start()
+    }
+
+    /**
+     * Gets the MediaPlayer that plays the 'correct' feedback (eg. 'bravo!')
+     * ONLY USE THIS FOR UNIT TESTING
+     */
+    fun __testing_getCorrectMP(): MediaPlayer = correctMp
+    /**
+     * Gets the MediaPlayer that plays the 'incorrect' feedback (eg. 'domamge!')
+     * ONLY USE THIS FOR UNIT TESTING
+     */
+    fun __testing_getIncorrectMP(): MediaPlayer = incorrectMp
```

Comment:
I don't know if it's a good way of doing things to put functions only for testing like this (Even if I don't see another way of doing it). For my knowledge, is this a good way of doing?

Reply:
yes, this is somewhat bad, but I really didn't see a way to make meaningful tests without it. But the `__testing` prefix at least clearly indicates that it should not be used elsewhere

Reply:
okay okay, I think that it is clear enough. Okay for me!